addons/point_of_sale/static/src/app/service_worker.js

```diff
@@ -0,0 +1,44 @@
+// @odoo-module ignore
+/* eslint-disable no-restricted-globals */
+
+const cacheName = "odoo-pos-cache";
+
+const cacheResources = async (url) => {
+    try {
+        const cache = await caches.open(cacheName);
+        await cache.add(url);
+    } catch (error) {
+        console.info("Failed to cache resource", url, error);
+    }
+};
+
+const getFromCache = async (url) => {
+    const cache = await caches.open(cacheName);
+    return cache.match(url);
+};
+
+self.addEventListener("fetch", async (event) => {
+    const url = event.request.url;
+
+    // Ignore Chrome extensions and dataset. Dataset will be cached in indexedDB.
+    if (url.includes("extension") || url.includes("web/dataset")) {
+        return;
+    }
+
+    event.respondWith(
+        getFromCache(url).then((response) => {
+            if (response) {
+                return response;
+            }
+
+            console.warn("Serving from network", url);
+            return fetch(event.request);
+        })
```

Comment:
Shouldn't we prioritize the network? The cache should serve as backup.

Reply:
Its a PoC for now, i'm just trying to make it works, after i will adapt that kind of stuff